segmentation/utils/spatial_analysis.py

```diff
@@ -63,47 +67,138 @@ def calculate_channel_spatial_enrichment(dist_matrix, marker_thresholds, all_pat
+    cell types, looking for both positive and negative enrichment. Uses bootstrapping to permute cell labels randomly.
+
+    Args:
+        all_data: data including points, cell labels, and
+            cell expression matrix for all markers
+        dist_mats: A dictionary that contains a cells x cells matrix with the euclidian
+            distance between centers of corresponding cells for every fov
+        fovs: patient labels to include in analysis. If argument is none, default is all labels used
+        bootstrap_num: number of permutations for bootstrap. Default is 1000
+        dist_lim: cell proximity threshold. Default is 100
+
+    Returns:
+        values: a list with each element consisting of a tuple of
+            closenum and closenumrand for each point included in the analysis
+        stats: an Xarray with dimensions (points, stats, number of markers, number of markers) The included stats
+            variables are:
+            z, muhat, sigmahat, p, h, adj_p, and marker_titles for each point in the analysis"""
+
+    fov_col = "SampleID"
+    cell_type_col = "cell_type"
+    flowsom_col = "FlowSOM_ID"
+    cell_label_col = "cellLabelInImage"
+
+    # Setup input and parameters
+    if fovs is None:
+        fovs = list(set(all_data[fov_col]))
+        num_fovs = len(fovs)
+    else:
+        num_fovs = len(fovs)
+
+    values = []
+
+    # Error Checking
+    if not np.isin(fovs, all_data[fov_col]).all():
+        raise ValueError("Points were not found in Expression Matrix")
+
+    # Extract the names of the cell phenotypes
+    pheno_titles = all_data[cell_type_col].drop_duplicates()
+    # Extract the columns with the cell phenotype codes
+    pheno_codes = all_data[flowsom_col].drop_duplicates()
+    # Get the total number of phenotypes
+    pheno_num = len(pheno_codes)
+
+    # Subset matrix to only include the columns with the patient label, cell label, and cell phenotype
+    fov_cluster_data = all_data[[fov_col, cell_label_col, flowsom_col]]
+
+    # Create stats Xarray with the dimensions (points, stats variables, number of markers, number of markers)
+    stats_raw_data = np.zeros((num_fovs, 7, pheno_num, pheno_num))
+    coords = [fovs, ["z", "muhat", "sigmahat", "p_pos", "p_neg", "h", "p_adj"], pheno_titles, pheno_titles]
+    dims = ["points", "stats", "pheno1", "pheno2"]
+    stats = xr.DataArray(stats_raw_data, coords=coords, dims=dims)
+
+    for i in range(0, len(fovs)):
+        # Subsetting expression matrix to only include patients with correct label
+        patient_ids = fov_cluster_data.iloc[:, 0] == fovs[i]
+        fov_data = fov_cluster_data[patient_ids]
+
+        # Subset the distance matrix array to only include the distance matrix for the correct point
+        # dist_mat = dist_mats[i, :, :].values
+        # distmat_fov_id = dist_mat_list == fovs[i]
+        # distmat_fov = [x for x in dist_mat_list if str(fovs[i]) in x][0]
```

Comment:
Okay, it seems like the code is a bit more finalized now. Can you go through and delete all of the commented out code throughout all of the different files?